pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -368,6 +372,7 @@ public void acknowledgeMessage(List<Position> positions, AckType ackType, Map<St
         }
 
         if (!cursor.getMarkDeletedPosition().equals(previousMarkDeletePosition)) {
+            deleteTransactionMarker((PositionImpl) cursor.getMarkDeletedPosition(), ackType, properties);
```

Comment:
The mark delete position of the cursor may be modified, it's better to avoid to all `cursor.getMarkDeletedPosition()`  twice, it might return a different position. So you can use a local variable to maintain the currentMarkDeletePosition.